recipes/libx264/all/conanfile.py

```diff
@@ -40,80 +50,94 @@ def configure(self):
-                args.append('--extra-ldflags=-arch arm64')
-
-            if tools.cross_building(self.settings):
-                if self.settings.os == "Android":
-                    # the as of ndk does not work well for building libx264
-                    self._override_env["AS"] = os.environ["CC"]
-                    ndk_root = tools.unix_path(os.environ["NDK_ROOT"])
-                    arch = {'armv7': 'arm',
-                            'armv8': 'aarch64',
-                            'x86': 'i686',
-                            'x86_64': 'x86_64'}.get(str(self.settings.arch))
-                    abi = 'androideabi' if self.settings.arch == 'armv7' else 'android'
-                    cross_prefix = "%s/bin/%s-linux-%s-" % (ndk_root, arch, abi)
-                    args.append('--cross-prefix=%s' % cross_prefix)
-
-            if self._is_msvc:
-                self._override_env['CC'] = 'cl'
-            self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-            if self._is_msvc:
-                self._autotools.flags.append('-%s' % str(self.settings.compiler.runtime))
-                # cannot open program database ... if multiple CL.EXE write to the same .PDB file, please use /FS
-                self._autotools.flags.append('-FS')
-            build_canonical_name = None
-            host_canonical_name = None
-            if self.settings.compiler == "Visual Studio":
-                # autotools does not know about the msvc canonical name(s)
-                build_canonical_name = False
-                host_canonical_name = False
-            self._autotools.configure(args=args, vars=self._override_env, configure_dir=self._source_subfolder, build=build_canonical_name, host=host_canonical_name)
-
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        args = [
+            "--bit-depth=%s" % str(self.options.bit_depth),
+            "--disable-cli",
+            "--prefix={}".format(tools.unix_path(self.package_folder)),
+        ]
+        if self.options.shared:
+            args.append("--enable-shared")
+        else:
+            args.append("--enable-static")
+        if self.options.get_safe("fPIC", self.settings.os != "Windows"):
+            args.append("--enable-pic")
+        if self.settings.build_type == "Debug":
+            args.append("--enable-debug")
+        if self.settings.os == "Macos" and self.settings.arch == "armv8":
+            # bitstream-a.S:29:18: error: unknown token in expression
+            args.append("--extra-asflags=-arch arm64")
+            args.append("--extra-ldflags=-arch arm64")
+
+        if self._with_nasm:
+            # FIXME: get using user_build_info
+            self._override_env["AS"] = os.path.join(self.deps_cpp_info["nasm"].rootpath, "bin", "nasm{}".format(".exe" if tools.os_info.is_windows else "")).replace("\\", "/")
```

Comment:
Is it the fix for MinGW?

Reply:
The error was:
```
----Running------
> C:\.conan\36de63\1\bin\msys64\usr\bin\bash.exe --login -c ^"cd \^"/c/users/maarten/.conan/data/libx264/20191217/_/_/build/526e61ab394798c6e72c93aafa8d2122e49dc8fd\^" ^&^& PATH=\^"/c/users/maarten/.conan/data/mingw-w64/8.1/_/_/package/6903a9d1b48b06f7fbe0929db654512c77e6cc32/bin:/c/users/maarten/.conan/data/nasm/2.15.05/_/_/package/0a420ff5c47119e668867cdb51baff0eca1fdb68/bin:/c/.conan/36de63/1/bin/msys64/usr/bin:$PATH\^" ^&^& /c/users/maarten/.conan/data/libx264/20191217/_/_/build/526e61ab394798c6e72c93aafa8d2122e49dc8fd/source_subfolder/configure --disable-cli --prefix=/c/users/maarten/.conan/data/libx264/20191217/_/_/package/526e61ab394798c6e72c93aafa8d2122e49dc8fd --enable-static --bit-depth=all  ^"
-----------------
Found GNU assembler (GNU Binutils) 2.30
Minimum version is nasm-2.13
If you really want to compile without asm, configure with --disable-asm.
```
I noticed the version of nasm added as a build requirement was sufficient recent.
So the configure script must have failed to detect it.